hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/CalciteSqlOptimizer.java

```diff
@@ -352,13 +353,27 @@ private SqlPlan toCreateMappingPlan(PlanKey planKey, SqlCreateMapping sqlCreateM
         List<MappingField> mappingFields = sqlCreateMapping.columns()
                 .map(field -> new MappingField(field.name(), field.type(), field.externalName()))
                 .collect(toList());
-        Mapping mapping = new Mapping(
-                sqlCreateMapping.nameWithoutSchema(),
-                sqlCreateMapping.externalName(),
-                sqlCreateMapping.type(),
-                mappingFields,
-                sqlCreateMapping.options()
-        );
+
+        Mapping mapping;
+        if (nodeEngine.getVersion().asVersion().isLessThan(V5_3)) {
+            mapping = new Mapping(
+                    sqlCreateMapping.nameWithoutSchema(),
+                    sqlCreateMapping.externalName(),
+                    sqlCreateMapping.type(),
+                    mappingFields,
```

Comment:
you silently ignore `dataLink` and `objectType`. If we want to support creating V5.2 Mapping (see comment above), then we should throw if someone tries to create mapping with those settings when the node/cluster is not upgraded to 5.3